lib/Sema/CSSimplify.cpp

```diff
@@ -3748,7 +3748,9 @@ bool ConstraintSystem::repairFailures(
       // a property with a function type or a method reference,
       // e.g. `foo.bar = 42` neither can be used if the destination
       // is not l-value.
-      if (!getType(destExpr)->is<LValueType>() && rhs->is<FunctionType>()) {
+      auto destType = getType(destExpr);
+      if (!destType->is<TypeVariableType>() && !destType->is<LValueType>() &&
```

Comment:
I think we could go one step farther here and check for destination being l-value or a type variable which _could be_ bound to an l-value otherwise it would be okay to record a fix since the solver already knows that type variable representing destination cannot possibly be l-value.

Reply:
Sure, just adjusted :)

Reply:
Sorry I might not been clear, what I really meant is that destination is either `isa<LValueType>()` or it's a type variable which `canBindToLValue()`..

Reply:
Sorry, was my misunderstanding...  changes made :)